frontend/src/models/dashboardsModel.js

```diff
@@ -0,0 +1,124 @@
+import { kea } from 'kea'
+import { router } from 'kea-router'
+import api from 'lib/api'
+import { delay, idToKey } from 'lib/utils'
+import React from 'react'
+import { toast } from 'react-toastify'
+
+export const dashboardsModel = kea({
+    actions: () => ({
+        delayedDeleteDashboard: id => ({ id }),
+        setLastVisitedDashboardId: id => ({ id }),
+    }),
+    loaders: () => ({
+        rawDashboards: [
+            {},
+            {
+                loadDashboards: async () => {
+                    const { results } = await api.get('api/dashboard')
+                    return idToKey(results)
+                },
+            },
+        ],
+        // We're not using this loader as a reducer per se, but just calling it `dashboard`
+        // to have the right payload ({ dashboard }) in the Success actions
+        dashboard: {
+            addDashboard: async ({ name }) => await api.create('api/dashboard', { name }),
+            restoreDashboard: async dashboard => await api.create('api/dashboard', dashboard),
+            renameDashboard: async ({ id, name }) => await api.update(`api/dashboard/${id}`, { name }),
+            deleteDashboard: async id => {
+                const dashboard = await api.get(`api/dashboard/${id}`)
+                await api.delete(`api/dashboard/${id}`)
```

Comment:
Why did you go with the hard delete pattern here, instead of just setting `delete` to true and then hiding those from the request? That way we can always get them back in the future, reduces complexity and is consistent with the rest of the app.